### PR TITLE
Limit access policy ops to routable viewsets

### LINKED ISSUE #3080: pulp rpm content upload fails with "add_for_object_creator not registered" error
**Version**
```
  "versions": [
    {
      "component": "core",
      "version": "3.20.0",
      "package": "pulpcore"
    },
    {
      "component": "rpm",
      "version": "3.17.9",
      "package": "pulp-rpm"
    }
  ],
```

**Describe the bug**
When trying to upload new rpm content, upload fails with following error:
```
# pulp rpm content upload --file /etc/pulp/scrape/splunk/splunkforwarder-9.0.0.1-9e907cedecb1-linux-2.6-x86_64.rpm --relative-path splunkforwarder-9.0.0.1-9e907cedecb1-linux-2.6-x86_64.rpm
Uploading file /etc/pulp/scrape/splunk/splunkforwarder-9.0.0.1-9e907cedecb1-linux-2.6-x86_64.rpm
.............................................Upload complete. Creating artifact.
Error: {"detail":"Creation hook 'add_for_object_creator' was not registered for this view set."}
```

**To Reproduce**
Install pulpcore with pulp_rpm plugin and try to upload rpm content

**Expected behavior**
No error is emitted

**Additional context**
We did upgrade from pulp 3.17.3 to 3.20.0 . It seems that the mentioned `add_for_object_creator` was removed in https://github.com/pulp/pulpcore/commit/4d50a746a30ba95ffab2dd3fbbc8d72df2c50c48 but maybe some migration scripts must be adjusted to ensure smooth upgrade with removal of that `add_for_object_creator` hook?

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #3080

### CODE REVIEW

**Review by gerrod3 — COMMENTED** [review]
Question I have from working on the Namespace migration in pulp_ansible, how do we allow outside endpoints to register their access policy? I was adding the new Namespace viewset only in `galaxy/v3` and found that the `AutoAddObjPermMixin` and AP migration didn't work since it couldn't find the viewset loaded from the `PluginConfig`. So instead I moved the new viewset over to `pulp_ansible/viewsets` and made its `routable = False` so it wouldn't show up in the Pulp APIs. This change would break my fix(/hack), so I was wondering if we had a proper way for outside Pulp endpoints.

**Comment by mdellweg** [comment]
Can you maybe overwrite the path for routing and keep routing=True?

**Comment by mdellweg** [comment]
As we cannot see how this really solves the original issue, i'm going to strip this pr down to the reset fix.

File: pulpcore/app/apps.py [hunk 1]
```diff
@@ -245,15 +245,19 @@ def _populate_access_policies(sender, apps, verbosity, **kwargs):
...
-                        print(
-                            "Access policy for {viewset_name} updated.".format(
-                                viewset_name=viewset_name
+                        if getattr(db_access_policy, key, None) != value:
+                            setattr(db_access_policy, key, value)
+                            dirty = True
+                    if dirty:
+                        db_access_policy.save()
+                        if verbosity >= 1:
+                            print(
```
Comment (gerrod3) [inline pulpcore/app/apps.py:256]: Nice!

**Review by gerrod3 — APPROVED** [review]

File: CHANGES/3080.bugfix [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Limited access policy reset to viewsets with a default one. This will solve 500 errors when
+trying to reset an access policy whose viewset name is repeated by an abstract base class.
+Stopped reporting on unmodified access policies when migrating.
```
Comment (bmbouter) [inline CHANGES/3080.bugfix:3]: Does it make sense to split this change into two bugfixes? I'm ok to keep it as one, but from a technical level each changelog should just be one change.

Comment (mdellweg) [reply CHANGES/3080.bugfix:3]: I think they are related enough to be on the same PR.

File: pulpcore/app/apps.py [hunk 3]
```diff
@@ -245,15 +245,19 @@ def _populate_access_policies(sender, apps, verbosity, **kwargs):
                             )
                         )
                 if not created and not db_access_policy.customized:
+                    dirty = False
```
Comment (bmbouter) [inline pulpcore/app/apps.py:248]: Consider maybe a name like `db_access_policy_out_of_date`? Just an idea. `dirty` also works for me. I generally try to come up with names that are a bit more self-descriptive.

**Review by bmbouter — APPROVED** [review]
LGTM but I'll leave the merging to @mdellweg in case he wants to adjust the `dirty` name or not (up to him). Thank you!

File: pulpcore/app/apps.py (hunk 3 above)
Comment (mdellweg) [reply pulpcore/app/apps.py:248]: I think "dirty" is quite a common name to signify that an ORM model is modified but not yet updated in the db.
It's just not built into Django.

**Pull request merged**